gridstatus/ercot.py

```diff
@@ -717,17 +749,203 @@ def _read_html_display(self, url, verbose=False):
 
         return df
 
-    def _get_supply_demand_json(self, verbose=False):
+    @support_date_range(frequency=None)
+    def get_hourly_load_post_settlements(
+        self,
+        date: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp],
+        end: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp] | None = None,
+        verbose: bool = False,
+    ) -> pd.DataFrame:
+        """Get historical hourly load data from ERCOT's load archives.
+
+        Downloads zip files from https://www.ercot.com/gridinfo/load/load_hist
+        and parses the historical load data by weather zones.
+
+        Arguments:
+            date (str, datetime): Year to download data for
+            end (str, datetime): Not used for this method
+            verbose (bool, optional): print verbose output. Defaults to False.
+
+        Returns:
+            pandas.DataFrame
+        """
+        if date == "latest":
+            return self.get_hourly_load_post_settlements("today", verbose=verbose)
+
+        date = utils._handle_date(date, self.default_timezone)
+        end = utils._handle_date(end, self.default_timezone)
+        start_year = date.year
+        end_year = end.year if end else start_year
+
+        logger.info(
+            f"Fetching historical load data for years {start_year} to {end_year}",
+        )
+
+        all_dfs = []
+        for year in range(start_year, end_year + 1):
+            logger.info(f"Processing year {year}...")
+            df = self._download_post_settlements_load_file(year)
+            all_dfs.append(df)
+            logger.info(f"Successfully processed year {year}, got {len(df)} rows")
+        combined_df = pd.concat(all_dfs, ignore_index=True)
+
+        mask = (combined_df["Interval Start"] >= date) & (
```

Comment:
in general, i think it's okay if these methods return more data than requested if that corresponds to how the source data is. the caller can always filter it out if desired